### DISCUSSION TITLE
Can EDDI recognize my fleet carrier when I drop out of supercruise?

### DISCUSSION THREAD

**Discussion by BarWeinig** [EDCD/EDDI#2186] [Needs thought]
This is a feature-request. Or maybe more an "is it possible" request.

When dropping out of supercruise, EDDI does not seem to recognize a (mine or any others) fleet carrier. It seems to recognize the nearest body, station or settlement. I have been spending quite some time trying to fix this, but, even with the EDDI/VA logs, I can't seem to get it to work. I even delayed EDDI's setting of variables, hoping that might change things (you drop often over 10 km from the fleet-carrier), but that doesn't work either.

I would really love it, if someone who knows what to look for, and how the information is processed, could look into this. The practical use of `event.bodytype_invariant = 'Station'`, `eddi_context_station_name = 'QNY-7VJ'` and other such variables are just above my comprehension, if I can't just use what is already there...

I have added this log [(2021-05-29.zip)](https://github.com/EDCD/EDDI/files/6565403/2021-05-29.zip) of me, starting the game, undocking from FC, jumping to SC, turning around, dropping near my FC, docking at FC and closing the game. Not sure this is really helpful, but at least you can access it, if you see a need.

Sincerely, Bar Weinig.

What EDDI said while visiting the 2 fleetcarriers in the system I'm in and returning to my own (unrelated to the log-file):
![image](https://user-images.githubusercontent.com/59501868/120085807-8532c580-c0db-11eb-8f1f-d4c596dda78f.png)

**Comment by Tkael** [comment]
Unfortunately, the data we receive about your location and proximity at that point is quite limited. 
Here's the relevant journal snippet from your log:
```json
{ "timestamp":"2021-05-29T19:07:55Z", "event":"SupercruiseExit", "Taxi":false, "Multicrew":false, "StarSystem":"Ikenga", "SystemAddress":3932143194826, "Body":"Ikenga A 2 a", "BodyID":11, "BodyType":"Planet" }
```

No data is provided about your proximity to your fleet carrier (or even indicating that a fleet carrier is present). 

The "BodyType" for this event is sometimes "Station" and when it is then we can report your location near a station. In this case, however, insufficient data is provided.

**Reply by BarWeinig** [reply]
Yeah, I did figure out that part too. It is line 16097. It is followed by line 16195, which is the first line mentioning my FC after dropping out. But it has the entirely same timestamp as the dropping out of SC. So, EDDI knows about the FC at the same time (relatively speaking), but, because it is already triggered, that information still comes too late (?). 
(Q1) Could I slightly delay the default script of "Entering normal space" and write a script based on (one of) the later events with the same timestamp to insert before EDDI runs the default script?

Also, there is only one way to visit an FC: through SC. So, I must have targeted an FC (talking about e.g `if find(stationname, "-") = 3`, maybe followed by `&&` stationname-string-length is 7), (Q2) so if I do not switch target, EDDI can be reasonably curtain I'll drop out at an FC? Not sure if it might even be possible to double-check based on "distance from target" or anything.

I'm just spitballing here, but I think there must be some way to give EDDI the information it needs, even though ED just doesn't.

**Comment by Tkael** [comment]
Line 16195 is recording the state of variables previously set, including your carrier as the last docked location. This isn't new information that we can act on to conclude that you've arrived at your carrier.

Targeting stations, bodies, or similar is invisible to EDDI - there is no journal event that captures non-ship and non-starsystem targets.

Distance from target is also not available to EDDI. The only distance measurement we know about through the journals is the distance from the main star (and even that only when certain events occur like scanning a body or docking at a station).

**Reply by BarWeinig** [reply]
Darn! Ok, idea thoroughly busted. I was pretty bummed I couldn't identify FC-targeting information, but now I get it. Thank you (again)! This seems to be one more thing that is truly lacking in the FDev implementation of FCs...

I'll try to let this go ;)

You can mark this topic as answered.
